test/shared/TestConnection.cs

```diff
@@ -139,6 +139,31 @@ public void Dispose()
             }
         }
 
+        public Task WaitForConnectionClose()
```

Comment:
New tests where calling it before I added the 408 code. Right now nothing is calling it, but I'll change `KeepAliveTimeoutTests` to use it because it's cleaner than checking for an exception on a send/receive attempt after the timeout.